scripts/echoserver/main.go

```diff
@@ -1,18 +1,33 @@
 package main
 
-// A simple echo server.  It listens on a random port, prints that port, then
-// echos back anything sent to it.
+// A simple echo server that listens on the specified network (tcp4 or tcp6) and
+// port, prints the resulting port (since you can use 0 to get a random port),
+// then echos back anything sent to it.  This is to test counting applications
+// that use port forwarding; currently only JetBrains uses this method.
+// Example usage: go run ./scripts/echoserver tcp6 0 -Didea.vendor.name=JetBrains
 
 import (
 	"errors"
 	"fmt"
 	"io"
 	"log"
 	"net"
+	"os"
 )
 
 func main() {
-	l, err := net.Listen("tcp", "127.0.0.1:0")
+	network := os.Args[1]
+	var address string
+	switch network {
+	case "tcp4":
+		address = "127.0.0.1"
+	case "tcp6":
+		address = "[::]"
```

Comment:
`[::]` is "null address" which is equivalent to `0.0.0.0` in IPv4 and means bind to all interfaces

What you're looking for is `[::1]` which is the local loopback address